sos/collector/sosnode.py

```diff
@@ -762,12 +764,35 @@ def execute_sos_command(self):
         try:
             path = False
             checksum = False
+            config_file_arg = ''
+            if self.opts.node_config_file:
+                config_file_arg = f'--config-file={self.opts.node_config_file}'
+            elif self.opts.inherit_config_file:
+                if not self.local:
+                    remote_config = os.path.join(self.tmpdir,
+                                                 'remote_sos.conf')
+                    self.run_command(f"mkdir -pv {self.tmpdir}",
+                                     timeout=10)
+                    self._transport.copy_file_to_remote(
+                        self.opts.inherit_config_file,
+                        remote_config)
```

Comment:
I'm not a fan of creating directories on the remote node, but we do need a place to put them and we wouldn't have a tempdir at this point.

What about placing them in `/tmp/`? We then are closer to our "no changes will be made" (which, writing that out, perhaps we should talk about that within this context?) stance, and we don't introduce the potential for an exploit that causes us to write to arbitrary locations if someone gets crafty enough.

Reply:
Works for me. I wasn't really happy with the way I implemented the temp directory handling anyway, I just didnt see a simple path to leverage **TempFileUtil** class without drastically changing collector/report interaction.

I'll make this use a unique name in `/tmp/` instead.